Resync `webrtc` from WPT upstream

Ahmad-S792 opened the pull request:
#### 3951337db1137641a12c7e56eeb579f4b1ee5969
<pre>
Resync `webrtc` from WPT upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=298020">https://bugs.webkit.org/show_bug.cgi?id=298020</a>
<a href="https://rdar.apple.com/problem/159838613">rdar://problem/159838613</a>

Reviewed by Youenn Fablet.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/cb0485e0505240c7fa264b6d0e0f78165432674b">https://github.com/web-platform-tests/wpt/commit/cb0485e0505240c7fa264b6d0e0f78165432674b</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCCertificate.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCConfiguration-iceServers-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCConfiguration-iceServers.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-close-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-close.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-id-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-id.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer-negotiated.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer-negotiated.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob-negotiated.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob-negotiated.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob-negotiated.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-helper.js: Added.
(rtc_data_channel_send_close_test):
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string-negotiated.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string-negotiated.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string-negotiated.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannelEvent-constructor.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceCandidate-constructor-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceCandidate-constructor.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceTransport-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceTransport.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addIceCandidate-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addIceCandidate.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver-renegotiation.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver-renegotiation.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-getStats.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-getStats.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-helper.js:
(async createDataChannelPairWithLabel):
(async createDataChannelPair):
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-ontrack.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-ontrack.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-remote-track-mute.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpParameters-encodings-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getParameters.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getSynchronizationSources.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getSynchronizationSources.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpSender-getParameters.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCSctpTransport-maxMessageSize-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCSctpTransport-maxMessageSize.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/munge-dont-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/outbound-rtp-encoding-index.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/outbound-rtp-encoding-index.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/av1-profile-asymmetry.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/av1-profile-asymmetry.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/crypto-suite.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/direction-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/direction.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/h264-unidirectional-codec-offer.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/h265-loopback.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/h265-loopback.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/video-codecs.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/video-codecs.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/rtp-stats-lifetime.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/rtp-stats-lifetime.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/w3c-import.log:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional-expected.txt.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc/protocol/direction-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/webrtc/RTCIceTransport-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceTransport-expected.txt.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/webrtc/protocols/h265-loopback.https-expected.txt: Added.
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC-expected.txt: Added.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative-expected.txt: Added.
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/304141@main">https://commits.webkit.org/304141@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9806d62d4f357e2f74d704ca6ec7dba819879df2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134711 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7157 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45953 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142231 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/86640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/63f0299e-2736-4dce-8310-c3c4401dd53c) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7765 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/156/builds/7010 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/102955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/86640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3566745c-5b1f-4016-bf45-80134e5532a4) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137658 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5462 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120736 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83763 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/cacb1c06-6a7f-4195-a6b4-97dc2595d217) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/5309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-API-Tests-EWS "Waiting to run tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/2821 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114536 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38883 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144926 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6831 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/39464 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/111349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/6905 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/156/builds/7010 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111648 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28311 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117016 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60715 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6882 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/35201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/6685 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/6921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/6794 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

philn commented on LayoutTests/platform/glib/TestExpectations:
```diff
@@ -2765,6 +2765,9 @@ http/wpt/webrtc/rtcNetworkInterface.html [ Failure ]
 webkit.org/b/276587 webrtc/video-av1.html [ Pass Timeout ]
 
+# Crashes on WebKitGTK and WPE platform
+imported/w3c/web-platform-tests/webrtc/protocol/h265-loopback.https.html [ Skip ]
```
That looks more like Skia-related I'm afraid:
```

Thread 1 (Thread 0x7fdb269e4100 (LWP 375246)):
#0  0x00007fdb44087db8 in GrResourceCache::refAndMakeResourceMRU(GrGpuResource*) [clone .cold] () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#1  0x00007fdb44088293 in GrResourceCache::findAndRefScratchResource(skgpu::ScratchKey const&) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#2  0x00007fdb4408bdbc in GrResourceProvider::findAndRefScratchTexture(skgpu::ScratchKey const&, std::basic_string_view<char, std::char_traits<char> >) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#3  0x00007fdb44080873 in GrResourceAllocator::Register::Register(GrSurfaceProxy*, skgpu::ScratchKey, GrResourceProvider*) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#4  0x00007fdb44081e31 in GrResourceAllocator::findOrCreateRegisterFor(GrSurfaceProxy*) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#5  0x00007fdb44082f68 in GrResourceAllocator::planAssignment() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#6  0x00007fdb4405e53b in GrDrawingManager::flush(SkSpan<GrSurfaceProxy*>, SkSurfaces::BackendSurfaceAccess, GrFlushInfo const&, skgpu::MutableTextureState const*) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#7  0x00007fdb4405f568 in GrDrawingManager::flushSurfaces(SkSpan<GrSurfaceProxy*>, SkSurfaces::BackendSurfaceAccess, GrFlushInfo const&, skgpu::MutableTextureState const*) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#8  0x00007fdb3ff236e3 in WebCore::CoordinatedPlatformLayerBufferNativeImage::CoordinatedPlatformLayerBufferNativeImage(WTF::Ref<WebCore::NativeImage, WTF::RawPtrTraits<WebCore::NativeImage>, WTF::DefaultRefDerefTraits<WebCore::NativeImage> >&&, WTF::OptionSet<WebCore::TextureMapperFlags>, std::unique_ptr<WebCore::GLFence, std::default_delete<WebCore::GLFence> >&&) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#9  0x00007fdb3ff2398b in WebCore::CoordinatedPlatformLayerBufferNativeImage::create(WTF::Ref<WebCore::NativeImage, WTF::RawPtrTraits<WebCore::NativeImage>, WTF::DefaultRefDerefTraits<WebCore::NativeImage> >&&, std::unique_ptr<WebCore::GLFence, std::default_delete<WebCore::GLFence> >&&) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#10 0x00007fdb43343ccd in WebCore::ImageBufferSkiaAcceleratedBackend::prepareForDisplay() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#11 0x00007fdb426bf9ab in WebCore::HTMLCanvasElement::prepareForDisplay() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#12 0x00007fdb422c4c7d in WebCore::Document::prepareCanvasesForDisplayOrFlushIfNeeded() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#13 0x00007fdb42e8fcf6 in WebCore::Page::forEachRenderableDocument(WTF::Function<void (WebCore::Document&)> const&) const () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#14 0x00007fdb42ea6919 in WebCore::Page::doAfterUpdateRendering() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#15 0x00007fdb42ea7178 in WebCore::Page::updateRendering() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#16 0x00007fdb3dbc3db8 in WebKit::WebPage::updateRendering() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#17 0x00007fdb3dc50441 in WebKit::LayerTreeHost::flushLayers() [clone .part.0] () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#18 0x00007fdb3dc512c8 in WebKit::LayerTreeHost::layerFlushTimerFired() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#19 0x00007fdb3dc41072 in WTF::Detail::CallableWrapper<WTF::RunLoop::Timer::Timer<WebKit::LayerTreeHost>(WTF::Ref<WTF::RunLoop, WTF::RawPtrTraits<WTF::RunLoop>, WTF::DefaultRefDerefTraits<WTF::RunLoop> >&&, WTF::ASCIILiteral, WebKit::LayerTreeHost*, void (WebKit::LayerTreeHost::*)())::{lambda()#1}, void>::call() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#20 0x00007fdb3fe333ce in WTF::RunLoop::TimerBase::TimerBase(WTF::Ref<WTF::RunLoop, WTF::RawPtrTraits<WTF::RunLoop>, WTF::DefaultRefDerefTraits<WTF::RunLoop> >&&, WTF::ASCIILiteral)::{lambda(void*)#1}::_FUN(void*) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#21 0x00007fdb3fe33ebd in WTF::RunLoop::{lambda(_GSource*, int (*)(void*), void*)#1}::_FUN(_GSource*, int (*)(void*), void*) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#22 0x00007fdb3830249e in ??? () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#23 0x00007fdb38361737 in ??? () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#24 0x00007fdb38302f87 in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#25 0x00007fdb3fe34018 in WTF::RunLoop::run() () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#26 0x00007fdb3dc55b56 in int WebKit::AuxiliaryProcessMain<WebKit::WebProcessMainWPE>(int, char**) () at /home/buildbot-worker/WPE-WK2-Tests-EWS/build/WebKitBuild/WPE/Release/lib/libWPEWebKit-2.0.so.1
#27 0x00007fdb384181ca in __libc_start_call_main (main=main@entry=0x55ad611ee7f0 <main>, argc=argc@entry=4, argv=argv@entry=0x7ffee02f5268) at ../sysdeps/nptl/libc_start_call_main.h:58
#28 0x00007fdb3841828b in __libc_start_main_impl (main=0x55ad611ee7f0 <main>, argc=4, argv=0x7ffee02f5268, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffee02f5258) at ../csu/libc-start.c:360
#29 0x000055ad611ee725 in _start ()
```

Ahmad-S792 replied:
Will raise bug - already did for another.

philn commented on LayoutTests/platform/ios/imported/w3c/web-platform-tests/webrtc/protocols/h265-loopback.https-expected.txt:
```diff
@@ -0,0 +1,5 @@
+
```
That file is in the incorrect folder (should be `protocol` without s afaics).

Ahmad-S792 replied:
Good spot, will update.

youennf commented on LayoutTests/platform/ios/TestExpectations:
```diff
@@ -8262,6 +8262,10 @@ webkit.org/b/298557 [ Debug ] imported/w3c/web-platform-tests/scroll-to-text-fra
+imported/w3c/web-platform-tests/webrtc/protocol/h265-loopback.https.html [ Skip ]
+imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative.html [ Failure Pass ]
+imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC.html [ Failure Pass ]
+imported/w3c/web-platform-tests/webrtc/RTCIceTransport.html [ Failure ]
```
Given it is timing out, I would tend to skip it for now.

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
Resync `webrtc` from WPT upstream
https://bugs.webkit.org/show_bug.cgi?id=298020
rdar://problem/159838613

Reviewed by Youenn Fablet.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/cb0485e0505240c7fa264b6d0e0f78165432674b

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCCertificate.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCConfiguration-iceServers-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCConfiguration-iceServers.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-close-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-close.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-id-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-id.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer-negotiated.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer-negotiated.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-array-buffer.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob-negotiated.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob-negotiated.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob-negotiated.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-blob.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-helper.js: Added.
(rtc_data_channel_send_close_test):
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string-negotiated.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string-negotiated.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string-negotiated.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close-string.window.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-close.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-send.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCDataChannelEvent-constructor.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceCandidate-constructor-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceCandidate-constructor.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceTransport-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceTransport.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addIceCandidate-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addIceCandidate.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver-renegotiation.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver-renegotiation.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addTransceiver.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-getStats.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-getStats.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-helper.js:
(async createDataChannelPairWithLabel):
(async createDataChannelPair):
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-ontrack.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-ontrack.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-remote-track-mute.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpParameters-encodings-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getParameters.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getSynchronizationSources.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getSynchronizationSources.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpSender-getParameters.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCSctpTransport-maxMessageSize-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCSctpTransport-maxMessageSize.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/munge-dont-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/outbound-rtp-encoding-index.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/outbound-rtp-encoding-index.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/av1-profile-asymmetry.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/av1-profile-asymmetry.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/crypto-suite.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/direction-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/direction.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/h264-unidirectional-codec-offer.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/h265-loopback.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/h265-loopback.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/video-codecs.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/video-codecs.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/protocol/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/rtp-stats-lifetime.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/rtp-stats-lifetime.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc/w3c-import.log:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.optional-expected.txt.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/webrtc/protocol/direction-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/webrtc/RTCIceTransport-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCIceTransport-expected.txt.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/webrtc/protocols/h265-loopback.https-expected.txt: Added.
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/webrtc/RTCDataChannel-worker-GC-expected.txt: Added.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-track-settings.tentative-expected.txt: Added.
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/304141@main
